crates/client/rpc/src/lib.rs

```diff
@@ -943,48 +944,88 @@ where
             }
         }
 
+        fn extract_fee_and_contract_address(
+            events: &Vec<starknet_core::types::Event>,
+            sequencer_address: &Felt252Wrapper,
+        ) -> (FieldElement, Option<FieldElement>) {
+            let transfer_event_key = starknet_core::utils::get_selector_from_name("Transfer").unwrap();
+            let find_transfer_fee_event = events.iter().find_map(|event| {
+                event.keys.iter().find(|&&key| key == transfer_event_key)?;
+                event.data.get(1).and_then(|send_to| (*send_to == sequencer_address.0).then_some(()))?;
+                Some(event)
+            });
+
+            match find_transfer_fee_event {
+                Some(transfer_fee_event) => {
+                    let actual_fee = Felt252Wrapper(transfer_fee_event.data[2].clone()).0; // ony low bits, see execute_fee_transfer() why
+                    let contract_address = Felt252Wrapper(transfer_fee_event.data[0].clone()).0;
+                    (actual_fee, Some(contract_address))
+                }
+                None => (Default::default(), None), // transaction without fee
+            }
+        }
+
+        let events_converted: Vec<starknet_core::types::Event> =
+            events.clone().into_iter().map(event_conversion).collect();
+
+        let (actual_fee, contract_address) = extract_fee_and_contract_address(&events_converted, &sequencer_address);
+
         let receipt = match tx_type {
             mp_transactions::TxType::Declare => TransactionReceipt::Declare(DeclareTransactionReceipt {
                 transaction_hash,
-                actual_fee: Default::default(),
+                actual_fee,
                 finality_status: TransactionFinalityStatus::AcceptedOnL2,
                 block_hash,
                 block_number,
                 messages_sent: Default::default(),
-                events: events.into_iter().map(event_conversion).collect(),
+                events: events_converted,
                 execution_result,
             }),
             mp_transactions::TxType::DeployAccount => {
+                let contract_address = match contract_address {
+                    Some(contract_address) => contract_address,
+                    None => {
+                        let transfer_event_key = starknet_core::utils::get_selector_from_name("Transfer").unwrap();
+                        let address_in_other_event = events_converted.iter().find_map(|event| {
+                            event.keys.iter().find(|&&key| key != transfer_event_key)?;
+                            Some(Felt252Wrapper(event.from_address.clone()).0)
+                        });
+                        match address_in_other_event {
+                            Some(contract_address) => contract_address,
+                            None => Default::default(), // should we calculate it than?
+                        }
+                    }
+                };
```

Comment:
Why is this one treated differently?

Reply:
In case we have special transaction with disabled fee, i guess there would not be any transfer event, but we still need contract_address, so it's second search for that.

Reply:
Okay, I found PR #868, and it seems that the `disable-transaction-fee` feature is not quite ready to be used because most of the tests do not assume that it can be configured that way. 
So I deleted this additional search.